numpy/f2py/capi_maps.py

```diff
@@ -718,10 +718,7 @@ def modsign2map(m):
 
 def cb_sign2map(a, var, index=None):
     ret = {'varname': a}
-    if index is None or 1:  # disable 7712 patch
```

Comment:
What a strange way to disable a path! I tried to understand why it was done, but failed. It goes back to https://github.com/numpy/numpy/commit/3d36bf277fd19e3db66bdd0201338a3db6dddc77, but that doesn't seem to have any comments about why it was done. Anyway, at this point it probably makes most sense to just remove it, as you do here.

Reply:
The strange thing is that it makes the path unreachable - a path that doesn't seem all that illogical. But if `f2py` more or less worked since 2009, it seems best now to just assume we only need the single branch - to do better would need to investigate whether the function ever gets called multiple times with different values of `index`.

Reply:
We can ask @pearu, but I would be surprised if he can still recall anything about it.